src/Resolve/nodes.jl

```diff
@@ -61,14 +40,22 @@ $(TYPEDSIGNATURES)
 Resolves random search AbstractParam nodes and places parameter results in the vals dictionary.
 """
 function node(
-    item::Delayed.AbstractParam, vals::Trials.ValsDict, params::Dict{Symbol, Types.AbstractDelayed},
+    item::Delayed.AbstractParam, vals::Trials.ValsDict, nid::Symbol,
     trials::Vector{Trials.Trial}, config::Config
 )::Union{IndexObjects.IndexInt, Real}
     if haskey(vals, item.label)
         throw(KeyError("Key $(item.label) already present $(vals[item.label])"))
     end
 
-    obj = node(item.obj, vals, params, trials, config)
+    # Following condition is used to make sure that the vals name is the same as hyperparameter name
+    # but currently there are many unit tests which have different names.
+    # if nid == Symbol()
+    #     nid = item.label
+    # elseif nid != item.label
```

Comment:
Just wanted to clarify the commented out lines. The `nid` param is currently unused in this `node` function so there's only the type assertion via the signature whether `nid` is a Symbol, otherwise a `MethodError` will be thrown. Do you want to keep the `KeyError` you commented out?

Reply:
> Just wanted to clarify the commented out lines. The `nid` param is currently unused in this `node` function so there's only the type assertion via the signature whether `nid` is a Symbol, otherwise a `MethodError` will be thrown. Do you want to keep the `KeyError` you commented out?

Thanks for your comment. This is for the next ticket (if requires) to make sure that the vals name is the same as hyperparameter name, which also requires further changes in other locations.

Reply:
Probably it's better to annotate with `# TODO check vals name is the same as hyperparameter name` or add to the issues as a request. Do you have an example when `item.label` doesn't match `nid`?

Reply:
Many thanks, I will add the notes! 
There are many examples in the unit test as one below:
space = Dict(
    :u => HP.Uniform(:param1, 0.0, 5.0),
    :qu => HP.QuantUniform(:param2, 0.0, 5.0, 1.0),
    :lu => HP.LogUniform(:param3, 1.0, 10.0),
    :qn => HP.QuantNormal(:param4, 2.0, 0.25, 1.0),
)